user-management/application/user/user.go

```diff
@@ -0,0 +1,108 @@
+package user
+
+import (
+	"log"
+	usr "mradulrathore/onboarding-assignments/user-management/domain/user"
+	"sort"
+	"strings"
+)
+
+var users = []usr.User{}
+
+func Init(usrs []usr.User) {
+	users = usrs
+}
+
+func Insert(user usr.User) {
+	index := searchName(user)
+	insertAt(index, user)
+}
+
+// return the smallest index i in [0, n) at which user.Name should be inserted to maintain sorted list
+// if user.Name matches with already existing element name then return index according to rollno
+func searchName(user usr.User) (index int) {
+	sort.Search(len(users), func(i int) bool {
+		if strings.Compare(users[i].Name, user.Name) == 1 {
+			return true
+		} else if strings.Compare(users[i].Name, user.Name) == 0 {
+			return users[i].RollNo > user.RollNo
+		}
+		return false
+	})
+
+	return
+}
+
```

Comment:
user db analogy.

give proper name to file

Reply:
done!